Orange/data/io_util.py

```diff
@@ -107,9 +107,14 @@ def isnastr(arr, out=None):
     mask : np.ndarray
     """
     arr = np.asarray(arr)
-    if out is None and arr.shape != ():
-        out = np.empty_like(arr, dtype=bool)
-    return __isnastr(arr, out=out)
+    na = __isnastr(arr)
+    if out is not None:
+        out[:] = na
+        return out
+    elif arr.shape != ():  # na is an object array
+        return na.astype(bool)
+    else:  # na is not a numpy array
+        return na
```

Comment:
Maybe just add `casting="unsafe"` to `__isnastr` call? Optionally even check `np.can_cast('bool', out.dtype)`